unlock-app/src/components/interface/checkout/alpha/Checkout/utils.ts

```diff
@@ -0,0 +1,48 @@
+interface FetchRecipientsDataOptions {
+  recipients: string[]
+  network: number
+  lockAddress: string
+}
+
+export async function fetchRecipientsData(
+  url: string,
+  { lockAddress, network, recipients }: FetchRecipientsDataOptions
+) {
+  try {
+    const result: string[] = []
+    for (const recipient of recipients) {
+      const endpoint = new URL(url)
+
+      endpoint.searchParams.append('network', network.toString())
+      endpoint.searchParams.append('lockAddress', lockAddress)
+      endpoint.searchParams.append('recipient', recipient)
+
+      const abortController = new AbortController()
+
+      const timer = setTimeout(() => {
+        abortController.abort()
+      }, 5000)
+
+      const response = await fetch(endpoint, {
```

Comment:
Is this going to work in a context where the 3rd party site is on _any_ domain? and maybe even HTTP (no s)?

Reply:
proxied using our RPC endpoint so it should work in almost all cases.